File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = n
    real_lost = list(set(lost) - set(reserve))
    real_reserve = list(set(reserve) - set(lost))
    cnt = 0
    real_lost.sort()
    for l in real_lost:
        if l-1 in real_reserve:
            real_reserve.remove(l-1)
            cnt += 1
        elif l+1 in real_reserve:
            real_reserve.remove(l+1)
            cnt += 1
    return answer - len(real_lost) + cnt

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] count = new int[n+1];
        
        for(int l : lost){
            count[l]--;
        }
        for(int r : reserve){
            count[r]++;
        }
        
        int answer = 0;
        for (int i = 1; i <= n; i++) {
            if (count[i] >= 0) {
                answer++;
            } else {
                if (i > 1 && count[i - 1] >= 1) {
                    count[i - 1]--;
                    answer++;
                } else if (i < n && count[i + 1] >= 1) {
                    count[i + 1]--;
                    answer++;
                }
            }
        }
        return answer;
    }
}